### DISCUSSION TITLE
why it can not be matched Id

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1289] [Q&A] [answered]
in this below screenshot,i want to refer LdNode:ID as sourceId and targetId,but why it can not be matched,if i change to refer LdNode:NAME,it match well,the ID is a random digital(24948462 for example )
![image](https://github.com/eclipse-langium/langium/assets/26995040/e48a1a12-4cc9-4fb7-a91e-01f6a57c990b)

**Comment by msujew** [comment, marked as answer]
@zrwsmd In order to give each element a canonical name, Langium uses the `name` property by default. This `name` property is used by the default scoping to find the element in cross references.

You need to adjust your `NameProvider` service if you want to change that behavior for a specific node.

**Reply by zrwsmd** [reply]
i got it,thanks a lot
